src/tools/wasm-emscripten-finalize.cpp

```diff
@@ -75,11 +78,16 @@ int main(int argc, const char *argv[]) {
                                           const std::string &argument) {
              numReservedFunctionPointers = std::stoi(argument);
            })
-      .add("--global-base", "", "Where lld started to place globals",
+      .add("--global-base", "", "Where lld should start to place static globals",
            Options::Arguments::One,
            [&globalBase](Options*, const std::string&argument ) {
              globalBase = std::stoull(argument);
            })
+      .add("--stack-base", "", "Where the stack begins",
+           Options::Arguments::One,
```

Comment:
Do we need this new argument?  Won't the existing stack pointer assigned by lld be correct?

Reply:
lld's guess at the stack pointer location is incorrect due to static allocations from JS (see my last comment, may have been a race).

Emscripten won't use the stack from JS, as the wasm controls it - it needs to call into wasm to get the stack pointer and modify it. So it can modify it after the wasm is ready and the program started up.

Good idea to rename this to `--initial-stack-pointer`, done.

Reply:
My understanding is that the memory layout used by lld/wasm-backend is 

```
1.  zero page
2.  stack data
3.  bss data
4.  stack (grows down)
5.  heap 
```

I believe that emscripten adds global data after (3), which mean its clobber the end of the stack (since the stack growns down).    So I believe the inital stack pointer set by lld (which grows down) will always be correct... but I'm not totally sure about all this right now.

Happy for this change to land as is and we can try to simplify and possible remove this argument.

Reply:
Yes, emscripten adds more stuff after 3. But it keeps the stack size fixed while doing so, so the initial stack location becomes higher - it pushes up the entire range reserved for the stack, top and bottom.